packages/utils/src/puppeteer.ts

```diff
@@ -11,3 +13,102 @@ export function waitForPageError(page: puppeteer.Page): Promise<never> {
         });
     });
 }
+
+export function logConsoleMessages(page: puppeteer.Page) {
+    page.on('console', async (msg) => {
+        const msgArgs = await Promise.all(msg.args().map((a) => a.jsonValue()));
+        consoleLog(...msgArgs);
+    });
+}
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function loadTestPage(page: puppeteer.Page, testPageUrl: string, timeout: number) {
+  // This can keep the process from terminating for upto `timeout` if an error
```

Comment:
Remove this part: `But the problem should not occur in headless mode.`, I later discovered that it does occur in headless mode, just less frequently.